doc/rfcs/2023-08-17--ManagingVersions.md

```diff
@@ -0,0 +1,199 @@
+Managing Versions
+=================
+
+ - Initial Proposal: 2023-08-17
+ - Disposition: Under Review
+ - RFC PR: https://github.com/tock/tock/pull/3622
+
+Summary
+-------
+
+This document describes Tock's versioning approach as well as the
+development model for how we manage changes pending for next patch,
+minor, and major releases.
+
+
+Kernel Versioning
+-------
+
+We apparently never wrote down how our stability guarantees map to
+kernel versions, so we should do that. I tried to capture the essence of
+what we currently do in the new §2.1 and §2.2 for TRD104 attached to
+this PR.
+
+_Ideally, this is just writing down our existing policy, and is not too
+controversial._
+
+
+Master Branch Policy
+--------------------
+
+A fairly significant piece of the unwritten versioning rules is "what
+changes are allowed in master?"
+
+We have an answer from the 2.0 transition that we used: the default
+development branch should maintain ABI backwards compatibility but not
+necessarily forward compatibility.  I.e., in our versioning language,
+the master branch has 'next minor release' semantics.
+
+I think it's fair to say that we've basically stuck to that policy.
+I tried to capture this in the updates to the Maintenance.md document
+attached to this PR.
+
+_Ideally, this is just writing down our existing policy, and is not too
+controversial._
+
+I also tried to write down how we managed the major version transition.
+I think it worked pretty well last time, but open to more thoughts here.
+
+
+Understanding the delta between master and last release
+------------------------------------
+
+While we have a CHANGELOG file, to-date we are not great about keeping
+it up to date during active development. Rather, as part of release
+preparation the core team retroactively looks through all of the PRs and
+commit history since the last release and synthesizes the key changes.
```

Comment:
CHANGELOG takes work, and it's work no one wants to do (as evidenced by it not getting updated). Perhaps eventually we hire someone who can help. But I don't think any amount of documenting is going to get the current set of core members interested in maintaining a changelog.

This is fairly on-the-nose, forgive me, but rather than creating this PR, why not direct your "service" tock time towards maintaining the changelog? You created it (https://github.com/tock/tock/commit/a32a34de6ec9eebf21f47737f397e213413dbf3a) but I don't think have touched it since 2018, and I think it's reasonable to say you are the main person who remembers it is there.